extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/data_offer/DataOfferPageApiService.java

```diff
@@ -37,9 +59,52 @@ public IdAvailabilityResponse checkIfContractDefinitionIdAvailable(DSLContext ds
 
     private boolean isIdAvailable(DSLContext dsl, Table<?> table, TableField<?, String> idField, String id) {
         return !dsl.fetchExists(
-        dsl.select(idField)
-            .from(table)
-            .where(idField.eq(id))
-            );
+            dsl.select(idField)
+                .from(table)
+                .where(idField.eq(id))
+        );
+    }
+
+    public IdResponseDto createDataOffer(DSLContext dsl, DataOfferCreationRequest dataOfferCreationRequest) {
+        val commonId = dataOfferCreationRequest.getUiAssetCreateRequest().getId();
+
+        val assetIdExists = checkIfAssetIdAvailable(dsl, commonId).isAvailable();
+        if (!assetIdExists) {
+            throw new InvalidRequestException("Asset with id %s already exists".formatted(commonId));
+        }
+
+        val policyIdExists = checkIfPolicyIdAvailable(dsl, commonId).isAvailable();
+        if (!policyIdExists) {
+            throw new InvalidRequestException("Policy with id %s already exists".formatted(commonId));
+        }
+
+        val contractDefinitionIdExists = checkIfContractDefinitionIdAvailable(dsl, commonId).isAvailable();
```

Comment:
Would it make sense to inline those instead of allocating memory to constants?
Or is this for readability's sake?

Reply:
This is for readability, so the `/` is next to the name of what it negates.
I think it makes it easier to read than 

```java
if (!checkIfPolicyIdAvailable(dsl, commonId).isAvailable()) {
...
}
```

Reply:
I would inline it if I had something like the kotlin `Boolean.not()`